gtsam/base/Matrix.h

```diff
@@ -28,11 +28,15 @@
 #include <gtsam/base/Vector.h>
 #include <gtsam/config.h>
 
+#include <Eigen/Sparse>
```

Comment:
I have no time to do a thorough review until tomorrow, but can we do this PR without adding more includes in Matrix.h? Or in GaussianFactorGraph? We could maybe put all this new functionality in a new file, with most includes in .cpp. Include bloat significantly increases compilation time.

Reply:
done